File: backend/datatypes/membership.py
from typing import Optional
from enum import Enum
from pydantic import BaseModel


class Role(Enum):
    none = "none"
    player = "player"
    manager = "manager"
    root = "root"


class Membership(BaseModel):
    playerID: int
    teamID: Optional[int]
    role: Role

File: frontend/src/types/Membership.ts
import { type Role } from "./Role"

export type Membership = {
  teamID: number,
  playerID: number,
  role: Role
}

export const DEFAULT_MEMBERSHIP: Membership = {
  teamID: 0,
  playerID: 0,
  role: 'none'
}

export const DEFAULT_MEMBER_PLAYER: Membership = {
  teamID: 0,
  playerID: 0,
  role: 'player'
}

export const DEFAULT_MEMBER_MANAGER: Membership = {
  teamID: 0,
  playerID: 0,
  role: 'manager'
}